src/main/java/net/apnic/whowas/rdap/controller/RequestContext.java

```diff
@@ -0,0 +1,58 @@
+package net.apnic.whowas.rdap.controller;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.function.Supplier;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Utility class for RDAP endpoint to set the current thread local context for
+ * a request.
+ */
+public class RequestContext
+{
+    private final Supplier<HttpServletRequest> request;
+
+    public RequestContext(Supplier<HttpServletRequest> request)
+    {
+        this.request = request;
+    }
+
+    public URL getContext()
+    {
+        try
+        {
+            return new URL(request.get().getRequestURL().toString());
```

Comment:
I would suggest deleting this class, and passing the Supplier<HttpServletRequest> directly to the LinkSerialiser.

This is because the language / translation of this class, getRequestURL -> getContext only makes sense in the context of linking. It doesn't stand on its own and ought to be part of LinkSerialiser.

Reply:
I think its still serving a purpose. I don't want to have all the URL gen code in a link when thats not its purpose.

For the moment I am going to leave as is.